### ISSUE TITLE
[v0.3.1] Release Tracker

### ISSUE THREAD

**Issue by vmpuri** [pytorch/executorch#4602]
The v0.3.1 release will be cut from the "[release/0.3](https://github.com/pytorch/executorch/tree/release/0.3)" branch for critical fixes to the [v0.3.0](https://github.com/pytorch/executorch/releases/tag/v0.3.0) release. 

Branch Cut Date: 9/6/2024
Intended Release Date: 9/16/2024

This issue is for tracking cherry-picks to the release branch.

## Cherry-Pick Criteria

Focus on stability and documentation correctness. Ideally only low risk changes may be cherry-picked from main:

1. Critical fixes in core components: [build](https://github.com/pytorch/executorch/tree/main/build), [exir](https://github.com/pytorch/executorch/tree/main/exir), [backends](https://github.com/pytorch/executorch/tree/main/backends), [runtime](https://github.com/pytorch/executorch/tree/main/runtime), [sdk](https://github.com/pytorch/executorch/tree/main/sdk)
2. Bug fixes in demos/examples. No new features/experiments
3. Critical bug fixes based on community feedback
4. Documentation improvements
5. Test/CI fixes
6. Release branch specific changes (e.g. change version identifiers)

Any core components changes and other change will require special dispensation from the release managers (currently  @vmpuri  @dvorjackz). If this applies to your change please write "Special Dispensation" in the "Criteria Category:" template below and explain.

## Cherry-Pick Process

1. Ensure your PR has landed in master. This does not apply for release-branch specific changes (see Phase 1 criteria).
2. Create (but do not land) a PR against the release branch. 
   **Option 1**: Use `pytorchbot` to cherry pick a PR that has been committed to the main branch using `@pytorchbot cherry-pick` command as follows.
     
      ```
      @pytorchbot cherry-pick --onto release/0.3 -c {regression,critical,fixnewfeature,docs,release}
      ```
     
     <details>
     
      ```
      usage: @pytorchbot cherry-pick --onto ONTO [--fixes FIXES] -c
                               {regression,critical,fixnewfeature,docs,release}

      Cherry pick a pull request onto a release branch for inclusion in a release

      optional arguments:
        --onto ONTO           Branch you would like to cherry pick onto (Example: release/0.3)
        --fixes FIXES         Link to the issue that your PR fixes (i.e. https://github.com/pytorch/executorch/issues/2345)
        -c {regression,critical,fixnewfeature,docs,release}
                        A machine-friendly classification of the cherry-pick reason.
                        
      ```
      
      </details>
      
      **Option 2**: Manually create a cherry pick PR using git command as follows.
      <details>

      ```
      # Find the hash of the commit you want to cherry pick
      # (for example, abcdef12345)
      git log

      git fetch origin release/0.3
      git checkout release/0.3
      git cherry-pick -x abcdef12345

      # Submit a PR based against 'release/0.3' either:
      # via the GitHub UI
      git push my-fork

      # via the GitHub CLI
      gh pr create --base release/0.3
      
      ```
      
      </details>

3. Make a request below with the following format:

```
Link to landed trunk PR:
* 

Link to release branch PR:
* 

Criteria category and reasons:
* 
```

1. Someone from the release team will reply with approved / denied or ask for more information.
2. If approved, someone from the release team will merge your PR once the tests pass. **Do not land the release branch PR yourself.**

Please note HUD Link with branch CI status and link to the HUD to be provided here.
[HUD](https://hud.pytorch.org/hud/pytorch/executorch/release%2F0.3)

### Versions

0.3.1

**Comment by dbort** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/4633

Link to release branch PR:
* [PLACEHOLDER]

Criteria category and reasons:
* Fix docs
